src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1649,4 +1873,5 @@ def setup(middleware):
     asyncio.ensure_future(kmod_load())
     asyncio.ensure_future(middleware.call('pool.dataset.register_attachment_delegate',
                                           VMFSAttachmentDelegate(middleware)))
+    asyncio.ensure_future(middleware.call('vm.initialize_vms', 2))  # We use a short timeout here deliberately
```

Comment:
Why is this done twice: here and after system boot?

Reply:
We want to make sure that we have initialized vms on middlewared restart and on boot. For boot, the first call would fail as libvirtd isn't running yet. Ideally this first call should only be made when `system.ready` returns true. I'll update this to use that. Would you know of any reason where `system.ready` can't be trusted ? It checks for `.bootready` file and it should be there, i wonder why it wasn't their earlier.
For now, i'll update this to correctly reflect that behavior.